gateway/ln-gateway/src/bin/cln_extension.rs

```diff
@@ -81,13 +81,10 @@ pub struct Htlc {
 
 #[derive(Clone, Serialize, Deserialize, Debug)]
 pub struct Onion {
-    pub payload: String,
-    pub short_channel_id: u64,
+    #[serde(default)]
+    pub short_channel_id: Option<u64>,
     #[serde(deserialize_with = "as_fedimint_amount")]
     pub forward_msat: Amount,
```

Comment:
```
2023-01-24T00:18:39.224Z **BROKEN** plugin-ln_gateway-logging: htlc_accepted error missing field `forward_msat`
```

:shrug:

Reply:
I was trying without this field either now, but I think I got rate-limited now for too many failed payments.

Reply:
> ```
> 2023-01-24T00:18:39.224Z **BROKEN** plugin-ln_gateway-logging: htlc_accepted error missing field `forward_msat`
> ```
> 
> shrug

`cln-extension` is out of path for your testing. yes, our current version of `cln-plugn` used in `ln-gateway` (that you have deployed) will this this error if you expected `forward_msat` in onion

Reply:
~@dpc this log message seems pre-#1496? @okjodom is right, the code touched by this PR is not used anywhere.~ 

But it's a good point: why not remove `forward_msat` just like in #1496?

Reply:
It'd be a bigger change touching subscribe intercept grpc spec!

Proper fix which would save us a lot of code churn, should start with #1245.

So I'm with keeping that problematic field, or stashing this PR for a while then fix they whole type issue on one go

Reply:
Follow up: look into if `forward_msat` also has to be optional. Maybe make the fields only present on intermediate hops a separate struct inside an `Option`. Would be good to get feedback from CLN, which #1245 will help getting.